### PR TITLE
Migrate to package:web

### PR DESCRIPTION
**Pull request by raldhafiri**
Migrated to package:web, meaning that it now supports WASM!

### CODE REVIEW

File: motion_web/lib/src/web_sensors_interop.dart
Comment (mrcendre) [inline motion_web/lib/src/web_sensors_interop.dart:1]: That's great! 

Is there a reason for adding interop to Accelerometer and Magnetometer too, since they are not used in the project ? If not, I would consider removing them.

File: motion_web/lib/src/motion.dart [hunk 1]
```diff
@@ -51,9 +51,7 @@ class WebMotion extends MotionPlatform {
   FramesPerSecond _updateInterval = defaultUpdateInterval;
 
   /// A convenience getter for the 'frequency' option of sensors constructors, expressed in 'updates per second'.
-  int get _frequency =>
-      _updateInterval.inMicroseconds ~/
-      const Duration(seconds: 1).inMicroseconds;
+  int get _frequency => _updateInterval.inMicroseconds ~/ const Duration(seconds: 1).inMicroseconds;
```
Comment (mrcendre) [inline motion_web/lib/src/motion.dart:54]: Changes like this one are probably due to the formatting settings of your IDE. However when publishing a package, a special formatting is expected.

Please format the package by running `dart format .` at the project's root and update this PR. This will remove a lot of unnecessary changes.

File: motion_web/pubspec.yaml [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+version: 1.4.1
 homepage: https://cendre.me
 repository: https://github.com/mrcendre/motion
 issue_tracker: https://github.com/mrcendre/motion/issues?q=is%3Aissue+is%3Aopen
 
 environment:
-  sdk: ">=2.12.0 <3.0.0"
-  flutter: ">=1.20.0"
+  sdk: ">=3.3.0"
+  flutter: ">=3.19.0"
```
Comment (mrcendre) [inline motion_web/pubspec.yaml:10]: This change is actually a big deal, but I think it is acceptable because Flutter 3 has been around for long enough now. It will require to bump all packages to a new major version though, as these changes might be breaking.

I will take care of it as soon as #16 is merged.

**Review by mrcendre — CHANGES_REQUESTED** [review]
Thank you so much for your contribution @raldhafiri !

I caught a couple nits to address before merging. I will approve this PR as soon as they're fixed.

File: motion_web/lib/src/web_sensors_interop.dart
Comment (raldhafiri) [reply motion_web/lib/src/web_sensors_interop.dart:1]: issue resolved and interops removed
